prog/dns_zone/setup_dns_server_vm.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+class Prog::DnsZone::SetupDnsServerVm < Prog::Base
+  subject_is :vm, :sshable
+
+  def self.assemble(dns_server_id, name: nil, project_id: nil, vm_size: "standard-2", storage_size_gib: 30, location: "hetzner-fsn1")
+    unless (dns_server = DnsServer[dns_server_id])
+      fail "No existing Dns Server"
+    end
+
+    project_id ||= (Config.production? || Config.dns_service_project_id) ? Config.dns_service_project_id : Project.order_by(:created_at).first.id
+    unless (project = Project[project_id])
+      fail "No existing Project"
+    end
+
+    name ||= dns_server.ubid.to_s + SecureRandom.alphanumeric(8).downcase
+
+    DB.transaction do
+      vm_st = Prog::Vm::Nexus.assemble_with_sshable(
+        "ubi",
+        project.id,
+        location: location,
+        name: name,
+        size: vm_size,
+        storage_volumes: [
+          {encrypted: true, size_gib: storage_size_gib}
+        ],
+        boot_image: "ubuntu-jammy",
+        enable_ip4: true
+      )
+
+      Strand.create_with_id(prog: "DnsZone::SetupDnsServerVm", label: "configure_knot", stack: [{subject_id: vm_st.id, dns_server_id: dns_server_id}])
+    end
+  end
+
+  def ds
+    @ds ||= DnsServer[frame["dns_server_id"]]
+  end
+
+  label def configure_knot
+    nap 5 unless vm.strand.label == "wait"
+
+    register_deadline(nil, 15 * 60)
+
+    sshable.cmd <<~SH
+sudo sed -i 's/#DNSStubListener=yes/DNSStubListener=no/' /etc/systemd/resolved.conf
```

Comment:
AFAIR, we also need to specify alternative DNS resolvers such as Cloud9 or Cloudflare.

Reply:
Sorry, I meant Quad9. In any case, it seems this is not required. Normally after disabling systemd-resolved, DNS queries for external websites would stop working so we had to specify an external DNS server, but I just tested and it seems to work.